lib/vicadmin/validate.go

```diff
@@ -252,13 +255,25 @@ func (v *Validator) QueryDatastore(ctx context.Context, vch *config.VirtualConta
 		}
 	}
 
+	if sess.Client == nil {
+		return fmt.Errorf("Non-nil session govmomi client is nil while looking for datastore info")
+	}
+
+	if sess.Client.Client == nil {
+		return fmt.Errorf("Non-nil session vim client is nil while looking for datastore info")
+	}
+
 	pc := property.DefaultCollector(sess.Client.Client)
+	if pc == nil {
+		return fmt.Errorf("Could not get default propery collector; prop-collector came back nil")
+	}
+
 	err := pc.Retrieve(ctx, refs, nil, &dataStores)
```

Comment:
As we discussed in slack, the problem happens above, when looking up datastores with an invalid session.  The error is logged above
`log.Errorf("Unable to collect information for datastore %s: %s", dsName, err)`

But continues, and `refs` passed to `pc.Retrieve` has len == 0, causing the panic.  So the log.Errorf above should either be followed by a return error or you should check for `len(refs) == 0` here and avoid calling pc.Retrieve in that case.